workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -17,7 +26,52 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
   }
 
   public async addAccessPermission(params: AddRemoveAccessPermissionRequest): Promise<PermissionsResponse> {
-    throw new Error('Method not implemented.');
+    if (params.permission.accessLevel !== 'read-only' && params.permission.accessLevel !== 'read-write') {
+      throw new InvalidPermissionError('Access Level must be "read-only" or "read-write".');
+    }
+
+    if (params.permission.identityType !== 'GROUP' && params.permission.identityType !== 'USER') {
+      throw new InvalidPermissionError('IdentityType just be "GROUP" or "USER".');
+    }
+
+    const permissionsEffect: Effect = 'ALLOW';
+    const permissionsAction: Action = params.permission.accessLevel === 'read-only' ? 'READ' : 'UPDATE';
```

Comment:
Will `read-write` also have the `READ` action?

Reply:
okay. read-write now creates a READ and an UPDATE action. Note - it will also add a read-only access level in the return value along with the read-write as the READ implies this.